### PR TITLE
Add token and database URI validation to SQL Server configure script route

### PR DESCRIPTION
**Pull request by gabrielcorado**
Closes https://github.com/gravitational/security-findings/issues/70

Adds join token validation (same used on nodes join script route) and database URI validation.

In addition to the validations, this PR also changes the PowerShell script to use [single quotes](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-7.3#single-quoted-strings) (as there is no [need of string expansion](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-7.3#double-quoted-strings)).

### CODE REVIEW

File: lib/web/databases_test.go [hunk 1]
```diff
@@ -432,9 +432,19 @@ func TestHandleSQLServerConfigureScript(t *testing.T) {
 			},
 			assertError: require.Error,
 		},
+		{
```
Comment (jentfoo) [inline lib/web/databases_test.go:435]: Can you add test cases for other command injection characters:
* `;`
* `"`
* `'`
* `&`
* `|`
* `$`
* `\n`
* `()`
* `

File: lib/web/join_tokens.go [hunk 2]
```diff
@@ -436,6 +431,19 @@ func getJoinScript(ctx context.Context, settings scriptSettings, m nodeAPIGetter
...
 }
 
+// validateJoinToken validate a join token.
+func validateJoinToken(token string) error {
+	decodedToken, err := hex.DecodeString(token)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if len(decodedToken) != auth.TokenLenBytes {
+		return trace.BadParameter("invalid token %q", decodedToken)
```
Comment (jentfoo) [inline lib/web/join_tokens.go:441]: I would return a similar `BadParameter` error in the above condition too.

File: lib/web/databases_test.go (hunk 1 above)
Comment (gabrielcorado) [reply lib/web/databases_test.go:437]: @jentfoo I've added URI validation, and we're now escaping the URL when placing it on the PowerShell script.

The use cases you mentioned are split into two tests.

**Review by jentfoo — APPROVED** [review]
Unit tests failures look like they need attention, but generally looks good to me

**Review by smallinsky — APPROVED** [review]

**Review by greedy52 — APPROVED** [review]

**Pull request merged**